src/server/TCPServer.cpp

```diff
@@ -191,17 +218,63 @@ void TCPServer::HandlePrepareAck(const message::PrepareAck &m,
 
 void TCPServer::HandlePropose(const message::Propose &m,
     TCPConnection::pointer connection) {
-  round_->HandlePropose(m, connection);
+  auto round = GetRound(m.GetBallotNum());
+  round->HandlePropose(m, connection);
 }
 
 void TCPServer::TCPServer::HandleAccept(const message::Accept &m,
     TCPConnection::pointer connection) {
-  round_->HandleAccept(m, connection);
+  auto round = GetRound(m.GetBallotNum());
+  round->HandleAccept(m, connection);
 }
 
 void TCPServer::HandleLearn(const message::Learn &m,
     TCPConnection::pointer connection) {
-  round_->HandleLearn(m, connection);
+  auto round = GetRound(m.GetBallotNum());
+  round->HandleLearn(m, connection);
+}
+
+std::string TCPServer::Owner(int instance) {
+  auto server_addresses = Utilities::ReadConfig(kConfigFilePath);
+  int index = instance % num_servers_;
+  return std::get<2>(server_addresses[index]);
+}
+
+void TCPServer::CheckCommit() {
+  while (rounds_.find(expected_) != rounds_.end()) {
+    auto learned = rounds_[expected_]->GetLearnedValue();
+    if (learned->GetAction() != KVStore::Action::kNoOp) {
+      // Value is committed. Execute and return to client.
+      std::cout << "Committing value for round " << expected_ << std::endl;
+
+      auto amo_response = app_->Execute(*learned);
+      auto client_connection = clients_[expected_];
+      if  (client_connection) {
+        // If client connection is null, this value is being
+        // learned on a server that did not receive the request,
+        // so no need to send a reply.
+        auto response = message::Response(amo_response).Encode();
+        auto encoded = message::Message(response,
+            message::MessageType::kResponse).Encode();
+        Deliver(encoded, client_connection);
+      }
+    }
+
+    expected_++;
+  }
+}
+
+void TCPServer::OnLearned(int instance, KVStore::AMOCommand &value) {
+  std::cout << "CheckCommit, instance = " << instance << std::endl;
+
+  auto instance_owner = Owner(instance);
+  auto proposed_command = proposed_[instance];
+  if (instance_owner.compare(server_name_) != 0 &&
+      proposed_[instance] == value) {
```

Comment:
Use proposed_command for consistency

Reply:
Following pseudo-code variable names for fields.